File: 2-graph-fight/tester-solutions/main.cpp
#include<iostream>
using namespace std;
bool check(int degree[], int n) {
    int deg_sum = 0;
    for (int i = 0; i < n; i++)
        deg_sum += degree[i];
    return (2*(n-1) == deg_sum);
}
 
int main() {
    int n; cin>>n;
    int degree[n]; for(int i=0; i<n; i++) cin>>degree[i];
    if (check(degree, n)) cout << "Yes";
    else cout << "No";
    return 0;
}

File: 2-graph-fight/tester-solutions/main.py
def check(degree, n):
    deg_sum = sum(degree)
    return (2*(n-1) == deg_sum)
     
n = int(input())
degree = [int(item) for item in input().split()]
if (check(degree, n)):
    print("Yes")
else:
    print("No")